collect_app/src/test/java/org/odk/collect/android/projects/QrCodeProjectCreatorDialogTest.kt

```diff
@@ -72,6 +73,22 @@ class QrCodeProjectCreatorDialogTest {
         }
     }
 
+    // https://github.com/getodk/collect/issues/5266
+    @Test
+    fun `requestCameraPermission() should be called in onStart() to make sure it is called after returning to the dialog`() {
+        val scenario = launcherRule.launch(QrCodeProjectCreatorDialog::class.java)
```

Comment:
I just remembered that `FragmentScenarion` can have a specified initial state but our `FragmentScenarioLauncher#launch` doesn't support it yet. You could add that in here (see `FragmentScenarioLauncher#launchInContainer` for reference) and then you could just test that camera permissions are requested when moving to `STARTED` and not need to call `moveToState`. You'd also be able to just test that the permission is requested and that could be recorded with a boolean rather than a count which might be a bit nicer to read at the assertion points.

Reply:
Ok I've just added some improvements.